### PR TITLE
Use gsettings to select wallpaper

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
Hmm, as proposed this is GNOME specific (but easily customized). Should we spend any time on other options as there's a fallback?

File: glue/swaybg.launcher [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/bin/sh
 set -e
 
-background=$(snapctl get background)
+background=$(gsettings get org.gnome.desktop.background picture-uri | sed -Ee "s|[^:]*://(.*)'$|\1|")
 if [ -e "${background}" ]; then
-  export GDK_PIXBUF_MODULEDIR="$SNAP/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/gdk-pixbuf-2.0/2.10.0/loaders"
+  export GDK_PIXBUF_MODULEDIR="$SNAP/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders"
```
Comment (wmww) [inline glue/swaybg.launcher]: Why replace `SNAPCRAFT_ARCH_TRIPLET`? won't this fail on other architectures?

**Comment by wmww** [comment]
I guess it's fine if you never wand the background to be different from the GNOME one. For `mate-wayland` I ship my own background that I default to, which I like because it gives it a distinctive look.

File: glue/swaybg.launcher (hunk 1 above)
Comment (AlanGriffiths) [reply glue/swaybg.launcher]: That was an accident. Fixed

**Comment by AlanGriffiths** [comment]
> I guess it's fine if you never wand the background to be different from the GNOME one. For `mate-wayland` I ship my own background that I default to, which I like because it gives it a distinctive look.

Well, this is just a default. It is fairly easy to configure to use something else (by copying the `swaybg.launch` script and editing).

**Review by wmww — APPROVED** [review]
bors r+

**Pull request merged**